### PR TITLE
NoEmptyPhpdocFixer - do not leave empty line after docs removal

### PR DESCRIPTION
**Pull request by TomasVotruba**
As suggested here: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3574#issuecomment-373484579

Tests should fail now.

I can fix/complete tests, after initial review

### CODE REVIEW

**Comment by keradus** [comment]
we strongly follow tests driven development, hard to review without tests ;)

**Comment by TomasVotruba** [comment]
Nah. Just obvious fails on 10 lines :)

File: src/Fixer/Phpdoc/NoEmptyPhpdocFixer.php [hunk 1]
```diff
@@ -62,8 +63,23 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
...
+
+            $tokens->clearTokenAndMergeSurroundingWhitespace($index);
+
+            $previousToken = $tokens[$index - 1];
+            if ($previousToken->isWhitespace()) {
+                $previousWhitespaceContent = $previousToken->getContent();
+
+                $lastLineBreak = strrpos($previousWhitespaceContent, PHP_EOL);
+                $newWhitespaceContent = substr($previousWhitespaceContent, 0, $lastLineBreak);
+                if ($newWhitespaceContent) {
```
Comment (SpacePossum) [inline src/Fixer/Phpdoc/NoEmptyPhpdocFixer.php]: is this a test against `false` _or_ `0` by design? I think you want `false !==` here?

**Comment by SpacePossum** [comment]
@TomasVotruba  test results are in ;)

**Comment by julienfalque** [comment]
@TomasVotruba as @keradus requested, could you please add a test case that shows what you want to achieve here? That would be much helpful for reviewing indeed :)

**Comment by TomasVotruba** [comment]
Oh, I'm sorry but I'm too buzy ATM and don't have time for this :/

**Comment by SpacePossum** [comment]
@TomasVotruba did you find some time for this project to wrap this up? If not, no problem at all of course :)

**Comment by TomasVotruba** [comment]
Thanks for the ping :)

Nono, I have no motivation since working with tests on PHP CS Fixer is very confusing for me and I use own fixer for this.

**Comment by SpacePossum** [comment]
OK, I gave it a try but I'm still no fan of copy pasting the whole remove empty line logic into other rules because it becomes a mess. Anyway I drop the test cases here so I work on this later maybe;

```php
public function provideFixCases()
    {
        return [
            [
                '<?php
                    /** a */




                    /**
                     * test
                     */

                     /** *test* */
                ',
                '<?php
                    /**  *//** a *//**  */

                    /**
                    */

                    /**
                     *
                     */

                    /** ***
                     *
                     ******/

                    /**
**/
                    /**
                     * test
                     */

                     /** *test* */
                ',
            ],
            [
                '<?php #'
            ],
            [
                '<?php //'
            ],
            [
                '<?php /* */'
            ],
            [
                '<?php
                echo 1;
                echo 2;
                ',
                '<?php
                echo 1;
                /** */
                echo 2;
                ',
            ],
            [
                '<?php
                echo 3;


                echo 4;
                ',
                '<?php
                echo 3;

                /** */

                echo 4;
                ',
            ],
            [
                '<?php
echo 5;
echo 6;
                ',
                '<?php
echo 5;
/**  */
echo 6;
                ',
            ],
            [
                '<?php
echo 7; // 1  '.'
echo 8; // 2  '.'
',
                '<?php
echo 7; // 1  '.'
/**  */
echo 8; // 2  '.'
',
            ],
            [
                '<?php echo 9;  echo 10;',
                '<?php echo 9; /**  */ echo 10;',
            ],
            [
                '<?php echo 11;echo 12;',
                '<?php echo 11;/**  */echo 12;',
            ],
            [
                '<?php
echo 13; // 1
echo 14; // 2
',
                '<?php
echo 13; // 1
/**  */echo 14; // 2
',
            ],
            [
                '<?php
echo 131; // 1
 echo 141; // 2
',
                '<?php
echo 131; // 1
/**  */ echo 141; // 2
',
            ],
            [
                '<?php
                echo 101;
                echo 201;',
                '<?php
                echo 101;
                /**  */
                echo 201;',
            ],
            [
                '<?php
                echo 102;
                echo 202;
                ',
                '<?php
                echo 102;
                /**  *//**  *//**  *//**  */
                echo 202;
                ',
            ],
            [
                '<?php
                echo 1021;
                echo 2021;
                ',
                '<?php
                echo 1021;
            /**  *//**  *//**  *//**  */
/**  *//**  *//**  *//**  */
                        /**  *//**  *//**  *//**  */
                echo 2021;
                ',
            ],
            [
                '<?php
                echo 1022;
                ',
                '<?php
                echo 1022;
                /**  *//**  *//**  *//**  */',
            ],
            [
                '<?php
                echo "open tag fun 1";',
                '<?php
                /**  */
                echo "open tag fun 1";',
            ],
            [
                '<?php
                '.'
                '.'
                echo "open tag fun 1a";',
                '<?php
                /**  */
                '.'
                '.'
                echo "open tag fun 1a";',
            ],
            [
                '<?php
                echo "open tag fun 1b";',
                '<?php
                /**  */echo "open tag fun 1b";',
            ],
            [
                '<?php
                 echo "open tag fun 1c";',
                '<?php
                /**  */ echo "open tag fun 1c";',
            ],
            [
                '<?php '.'
                echo "open tag fun 2";',
                '<?php /**  */
                echo "open tag fun 2";',
            ],
            [
                '<?php
echo "open tag fun 2a";',
                '<?php
/**  */
echo "open tag fun 2a";',
            ],
        ];
    }
```

**Comment by TomasVotruba** [comment]
Why not push it to this branch so someone else could build on it?

**Comment by SpacePossum** [comment]
TBH I don't have the time to worry about multiple-user-PR's, so if I fixed it I prefer doing it on a new branch

**Comment by TomasVotruba** [comment]
I see

**Comment by SpacePossum** [comment]
closing in favor of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3961 and because the author of this PR stated to have no intentions to finish this PR

**Comment by TomasVotruba** [comment]
Of source, awesome :+1:  Thanks for continuing this

**Comment by SpacePossum** [comment]
thanks man, hope we'll see the processed and merged in :D

**Pull request closed without merging**